### PR TITLE
Fix touch events not dispatched to UIElement that have a RenderTransform

### PR DESCRIPTION
**Pull request by dr1rrb**
## PR Type
- Bugfix

## What is the current behavior?
Touch events are not dispatched to `UIElement` which has a `RenderTransform`

## What is the new behavior?
We can also tap on the part of a `UIElement`  which does not overlap it original location

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information
This is only a partial fix, cf. https://github.com/nventive/Uno/issues/649

Internal Issue (If applicable):
https://nventive.visualstudio.com/Umbrella/_workitems/edit/147978
https://nventive.visualstudio.com/Umbrella/_workitems/edit/148298
https://nventive.visualstudio.com/Umbrella/_workitems/edit/148300
https://nventive.visualstudio.com/Umbrella/_workitems/edit/148303
https://nventive.visualstudio.com/Umbrella/_workitems/edit/148375
https://nventive.visualstudio.com/Umbrella/_workitems/edit/148521

### CODE REVIEW

File: src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java [hunk 1]
```diff
@@ -298,7 +297,24 @@ public final void setAlpha(float opacity)
...
     }
 
-    public boolean dispatchTouchEvent(MotionEvent e)
+	/*
+	// To trace the 'dispatchTouchEvent', uncomment this and then uncomment logs in the method itself
+	private static String _indent = "";
+	public boolean dispatchTouchEvent(MotionEvent e)
+	{
+		String originalIndent = _indent;
+		Log.i("", _indent + "    + " + this.toString());
```
Comment (carldebilly) [inline src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java]: Is this costly?

File: src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java [hunk 2]
```diff
@@ -357,10 +373,72 @@ public boolean dispatchTouchEvent(MotionEvent e)
         // even if the point is outside the view bounds.
         _isPointInView = isLocalTouchPointInView(_transformedTouchX, _transformedTouchY); // takes clipping into account
 
-		// Always dispatch the touch events, otherwise system controls may not behave
-		// properly, such as not displaying "material design" animation cues (e.g. the
-		// growing circles in buttons when keeping pressed (RippleEffect)).
-		boolean superDispatchTouchEvent = super.dispatchTouchEvent(e);
+		//Log.i("", _indent + "MotionEvent: " + e.toString());
+		//Log.i("", _indent + "_isPointInView: " + _isPointInView);
```
Comment (carldebilly) [inline src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java]: commented code!

File: src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java [hunk 3]
```diff
@@ -387,16 +465,15 @@ public boolean dispatchTouchEvent(MotionEvent e)
...
-        // Log.i(this.toString(), "superDispatchTouchEvent: " + superDispatchTouchEvent);
-        // Log.i(this.toString(), "_childBlockedTouchEvent: " + _childBlockedTouchEvent);
-        // Log.i(this.toString(), "_childHandledTouchEvent: " + _childHandledTouchEvent);
-        // Log.i(this.toString(), "isBlockingTouchEvent: " + isBlockingTouchEvent);
-        // Log.i(this.toString(), "isHandlingTouchEvent: " + isHandlingTouchEvent);
+		//Log.i("", _indent + "superDispatchTouchEvent: " + superDispatchTouchEvent);
+        //Log.i("", _indent + "_childBlockedTouchEvent: " + _childBlockedTouchEvent);
+        //Log.i("", _indent + "_childHandledTouchEvent: " + _childHandledTouchEvent);
+        //Log.i("", _indent + "isBlockingTouchEvent: " + isBlockingTouchEvent);
+        //Log.i("", _indent + "isHandlingTouchEvent: " + isHandlingTouchEvent);
```
Comment (carldebilly) [inline src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java]: commented code!

File: src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java]: Oh yeah! (When dragging a `Slider` for instance)

File: src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java (hunk 3 above)
Comment (dr1rrb) [reply src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java]: Those are only to ease debugging as we are in java and we don't have a valid debugger nor even a real logging system. They were present and helped me a lot, I don't see the point to remove them.

File: src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java [hunk 4]
```diff
@@ -357,10 +373,72 @@ public boolean dispatchTouchEvent(MotionEvent e)
...
+		//Log.i("", _indent + "_isPointInView: " + _isPointInView);
+
+		// Note: Always dispatch the touch events, otherwise system controls may not behave
+		//		 properly, such as not displaying "material design" animation cues
+		//		 (e.g. the growing circles in buttons when keeping pressed (RippleEffect)).
+
+		boolean superDispatchTouchEvent = false;
+		if (_childrenTransformations.size() == 0
+			|| e.getAction() == MotionEvent.ACTION_CANCEL) {
+			// We don't any child which has s static transform, or the event is a "cancel" (cf. https://android.googlesource.com/platform/frameworks/base/+/0e71b4f19ba602c8c646744e690ab01c69808b42/core/java/android/view/ViewGroup.java#3013)
```
Comment (MatFillion) [inline src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java]: I think this sentence is missing a word.

**Pull request merged**